### DISCUSSION TITLE
What is the difference between filterable, queryable and searchable?

### DISCUSSION THREAD

**Discussion by best-cloud** [railsadminteam/rails_admin#3566] [Q&A] [answered]
Hi, I'm new to rails admin.

I don't find the definition of filterable, queryable and searchable in wiki, and I'm confused when implementing searching feature in list view.

Can someone explain what is the three xxxable used for?

Thank you so much.

**Comment by airled** [comment, marked as answer]
Hi there.

**Filterable**

`Filterable` defines fields that can be filtered with the menu in the top right corner.

![2023-03-01-132045_187x194_scrot](https://user-images.githubusercontent.com/10863999/222071936-657e6d54-b781-4e51-928c-0826ccc385f7.png)

By default all fields are filterable. But you can disable it by such:
```ruby
config.model 'Feed' do
  list do
    configure :name do
      filterable false
    end
  end
end
```
And now my field is gone:

![2023-03-01-132113_173x145_scrot](https://user-images.githubusercontent.com/10863999/222072196-79da52ac-6ffe-4c9d-9613-192e4dfd0527.png)

**Queryable**

`Queryable` defines fields, values of which can be searched via this search box:

![image](https://user-images.githubusercontent.com/10863999/222067479-732512c9-15a0-4a2b-a919-67690e050c42.png)

By default rails_admin searches in `id` and `name` fields. So if I type there `345` on my feed model page,

![image](https://user-images.githubusercontent.com/10863999/222068233-fcdd2d64-e225-4071-8969-234ff1a158e0.png)

rails_admin hits this sql query:
```
Feed Load (3.3ms)  SELECT "feeds".* FROM "feeds" WHERE ((feeds.id = 345) OR (feeds.name ILIKE '%345%')) ORDER BY feeds.id desc LIMIT $1 OFFSET $2  [["LIMIT", 20], ["OFFSET", 0]]
```

If I disable `name` field as queryable:
```ruby
config.model 'Feed' do
  list do
    configure :name do
      queryable false
    end
  end
end
```
rails_admin searches only in `id` values:
```
Feed Load (1.5ms)  SELECT "feeds".* FROM "feeds" WHERE ((feeds.id = 345)) ORDER BY feeds.id desc LIMIT $1 OFFSET $2  [["LIMIT", 20], ["OFFSET", 0]]
```

**Searchable**
If I get it right, searchable is just an alias for both queryable and filterable.
So
```ruby
configure :name do
  searchable false
end
```
is just an alias for

```ruby
configure :name do
  filterable false
  queryable false
end
```
But `searchable` also allows you to substitute one field with another or with a group of fields. You can read more [here](https://github.com/railsadminteam/rails_admin/wiki/List#fields-searching)
